test/src/686-azureblob_s3/main

```diff
@@ -0,0 +1,92 @@
+#
+# Test for running CVMFS against an Azure S3 implementation
+#
+# In this test we create our own Azurite blob service.
+# require az-cli and azurite to be installed, see install_azurite()
+# keys are hard-coded dev keys in azurite, so safe to have in code
+
+cvmfs_test_name="Azure blob on azurite"
+cvmfs_test_autofs_on_startup=false
+cvmfs_test_suites="quick"
+
+cleanup() {
+  echo "*** cleaning up"
+  sudo cvmfs_server rmfs -f azurite.cvmfs.io
+  if [ -f azurite_pid ]; then
+    sudo kill -9 $(cat azurite_pid)
+  fi
+}
+
+create_azurite_config() {
+  tee cvmfs_azurite.conf > /dev/null << EOF
+CVMFS_S3_HOST=127.0.0.1:10000
+CVMFS_S3_ACCESS_KEY=devstoreaccount1
+CVMFS_S3_SECRET_KEY=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==
+CVMFS_S3_BUCKET=devstoreaccount1/test
+CVMFS_S3_DNS_BUCKETS=false
+CVMFS_S3_FLAVOR=azure
+EOF
+}
+
+install_azurite() {
+  sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+  echo -e "[azure-cli]
+name=Azure CLI
+baseurl=https://packages.microsoft.com/yumrepos/azure-cli
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.microsoft.com/keys/microsoft.asc" | sudo tee /etc/yum.repos.d/azure-cli.repo
+  sudo yum install -y azure-cli
+
+  curl -fsSL https://rpm.nodesource.com/setup_16.x | sudo bash -
+  sudo yum install -y nodejs
+  sudo npm install -g azurite
+}
+
+start_azurite() {
+  echo "*** create azurite configuration"
+  create_azurite_config || return 20
+
+  echo "*** start azurite"
+  azurite_command="sudo azurite --debug azurite.txt --loose --skipApiVersionCheck --location ./"
+  local azurite_pid=$(run_background_service "$azurite_command")
+  echo $azurite_pid > azurite_pid
+  echo "*** Azurite PID is $azurite_pid"
+  if [ "x$azurite_pid" = "x" ]; then
+    return 10
+  fi
+
+  sleep 5
+
+  echo "*** create the container and we make it public"
+  az storage container create --name 'test' --connection-string 'DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;' --public-access blob || exit 21
+}
+
+cvmfs_run_test() {
+  # lets run this test only in the S3 suite
+  if [ x"$CVMFS_TEST_S3_CONFIG" == x"" ]; then
+    echo "not running, since s3 tests are blocked"
+    return 1
```

Comment:
We should exclude this test then in the non-S3 platforms in test/cloud_testing/platforms/$platform_test.sh files.

Reply:
excluded from all test files apart from centos7_s3